server/controllers/cohort.js

```diff
@@ -88,9 +103,64 @@ const updateCohortById = (req, resp) => {
   );
 };
 
+const getCohortData = (req, resp) => {
+  Cohort.findOne(
+    {
+      _id: req.params.id,
+      $or: [{ adminIds: req.user._id }, { memberIds: req.user._id }]
+    },
+    (err, doc) => {
+      if (err) {
+        return resp.status(400).send({ message: err.message });
+      }
+
+      if (!doc) {
+        return resp
+          .status(404)
+          .send({ message: 'Data not found for given cohort id!' });
+      }
+
+      const { _id, isArchived, name } = doc;
+
+      if (isArchived) {
+        return resp.status(200).json({ _id, isArchived, name });
+      }
+
+      return resp.status(200).json(doc);
+    }
+  );
+};
+
+const deleteCohortById = (req, resp) => {
+  let documentName = '';
+  Cohort.findOne({ _id: req.params.id, adminIds: req.user._id })
+    .then(doc => {
```

Comment:
what about error checking here?

Reply:
I used 'thens' instead of callbacks (err, doc) => {} because there are other async queries to the db in those controllers. All errors are captured by the catch method at the end of thens chain.